website/source/docs/agent/caching/index.html.md

```diff
@@ -0,0 +1,160 @@
+---
+layout: "docs"
+page_title: "Vault Agent Caching"
+sidebar_title: "Caching"
+sidebar_current: "docs-agent-caching"
+description: |-
+  Vault Agent Caching allows client-side caching of responses containing newly
+  created tokens and responses containing leased secrets generated off of these
+  newly created tokens.
+---
+
+# Vault Agent Caching
+
+Vault Agent Caching allows client-side caching of responses containing newly
+created tokens and responses containing leased secrets generated off of these
+newly created tokens. The renewals of the secrets that are cached are also
+managed by the agent.
+
+## Caching and Renewals
+
+Caching and renewals are managed by the agent only under these scenarios.
+
+1. Token creation requests are made through the agent and not directly to the
+   Vault server. Login endpoints from various auth methods also fall under this
+   category along with token creation endpoints of the token auth method.
+
+2. Leased secret creation requests are made through the agent using tokens that
+   are already managed by the agent.
+
+## Using Auto-Auth Token
+
+The requests that reach the agent, if they don't already bear a token as part
+of the request, the auto-auth token can be put to use. This feature is enabled
+by setting the `use_auto_auth_token`(see below) configuration field. Even if
+this configuration is set to `true`, if the requests that reach the agent
+already have a token attached on them, the attached token will be put to use
+instead of the auto-auth token.
+
+## Cache Evictions
+
+The eviction of cache entries will occur when the agent fails to renew secrets.
+This can happen when the secret that is cached hits it's maximum TTL or if the
+renewal results in an error.
+
+Agent also does some best-effort cache evictions by observing specific request
+types and response codes. For example, if a token revocation request is made
+via the agent and if the request succeeds, then agent evicts all the cache
+entries associated with the revoked token. Similar behavior is exercised for
+lease revocations as well.
+
+While agent tries to observe some requests and evicts cache entries
+automatically, it is possible that agent is completely unaware of revocations
```

Comment:
Isn't the agent completely unaware of revocations outside of the agent's context?  If so, remove "it is possible that".